executor/union_scan.go

```diff
@@ -156,7 +167,28 @@ func (us *UnionScanExec) Next(ctx context.Context, req *chunk.Chunk) error {
 			return nil
 		}
 		mutableRow.SetDatums(row...)
-		req.AppendRow(mutableRow.ToRow())
+
+		for _, idx := range us.virtualColumnIndex {
+			datum, err := us.schema.Columns[idx].EvalVirtualColumn(mutableRow.ToRow())
+			if err != nil {
+				return err
+			}
+			// Because the expression might return different type from
+			// the generated column, we should wrap a CAST on the result.
+			castDatum, err := table.CastValue(us.ctx, datum, us.columns[idx])
+			if err != nil {
+				return err
+			}
+			mutableRow.SetDatum(idx, castDatum)
+		}
+
+		matched, _, err := expression.EvalBool(us.ctx, us.conditionsWithVirCol, mutableRow.ToRow())
```

Comment:
why need this？Is it possible that `matched` is false and `err` is nil?

Reply:
This is a `selection` operator, used for `select a from t where a > 10`.
If one row has `a = 11`, this function will return `true`, if `a = 9`, return `false`.
We should filter all row meet the conditions, and discharge others rows.